wgsl/index.bs

```diff
@@ -5126,30 +5126,106 @@ If the `sampler_comparison` uses bilinear filtering then the returned value is
 the filtered average of these values, otherwise the comparision result of a
 single texel is returned.
 
-**TODO:**
 
-<pre class='def'>
-`vec4<type> textureLoad(texture_storage_ro_1d      , i32       coords)`
-`vec4<type> textureLoad(texture_storage_ro_2d      , vec2<i32> coords)`
-`vec4<type> textureLoad(texture_storage_ro_1d_array, vec2<i32> coords)`
-`vec4<type> textureLoad(texture_storage_ro_3d      , vec3<i32> coords)`
-`vec4<type> textureLoad(texture_storage_ro_2d_array, vec3<i32> coords)`
-  %32 = OpImageRead %v4float %texture_storage_ro %coords
+### `textureRead` ### {#textureread}
+
+Reads a single texel from a texture without sampling or filtering.
+
+<!-- TODO(ben-clayton): Remove parameter alignment padding -->
+
+```rust
+textureRead(t : texture_1d<T>,                    coords : u32)                                                           -> vec4<T>
```

Comment:
The discussion on coordinate types - #1257 - was closed recently, and if I understand correctly, the consensus is to keep `i32` for coordinates (matches my recommendation).

Reply:
I'm also a little uncertain what the final decision is there. I'd suggest landing this as i32, as that's consistent with the other functions, then creating a PR to change all atomically. Otherwise we'll have a weird intermediate state of signed and unsigned?

Reply:
Which other functions do you mean?

Reply:
`array_index`, `level` for depth textures

Reply:
My understanding is that array index and mipmap level are a different domain, i.e. the computations users could have to determine what those are would inherently produce non-negative results. I realize this vision may not be shared widely within the group, would appreciate the discussion to sync up on that on a call.

Reply:
I'm not convinced they're that wildly different domains. I've written shaders in the past that calculate array index and mipmap level derived from positional floating-point data. These could have easily have been unsigned, as much as a coordinate could.